connectors/sources/github.py

```diff
@@ -655,13 +655,11 @@ def get_rate_limit_encountered(self, status_code, message):
     async def _get_retry_after(self, resource_type):
         current_time = time.time()
         response = await self._get_client.getitem("/rate_limit")
-        reset = (
-            response.get("resources", {})
-            .get(resource_type, {})
-            .get("reset", current_time)
+        reset = nested_get_from_dict(
+            response, ["resources", resource_type, "reset"], default=current_time
         )
         # Adding a 5 second delay to account for server delays
-        return (reset - current_time) + 5
+        return (reset - current_time) + 5  # pyright: ignore
```

Comment:
What do we need to ignore here?

Reply:
pyright will complain `reset` can be `None` even though I have provided `current_time` as default